### PR TITLE
Cleanup project and ported Kotlin bugfix to Java version.

### PR DESCRIPTION
**Pull request by bodiam**
Apologies about the somewhat bigger PR, I got a bit too enthusiastic. 

- Removed Travis config
- Removed Kotlin dependencies
- Removed Gradle (use Maven instead)
- Removed Joiner (replaced by String.join)
- Cleanup code by using text blocks
- Added Maven Build Matrix (Java 17 and 21)

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -34,7 +34,7 @@ Also, currently known *not* supported items are:
 
 Add the following dependency to your project:
 
-`nl.jworks.markdown_to_asciidoc:markdown_to_asciidoc:1.0`
+`nl.jworks.markdown_to_asciidoc:markdown_to_asciidoc:2.0`
```
Comment (lefou) [inline README.adoc:37]: May I suggest to define an document attribute `version` (right below the document title) and using it here?

```suggestion
`nl.jworks.markdown_to_asciidoc:markdown_to_asciidoc:{version}`
```

Comment (bodiam) [reply README.adoc:37]: Seems like a good idea! Thanks for the suggestion!

**Review by dgautier — APPROVED** [review]
Seems Perfect :)

**Comment by bodiam** [comment]
In that case I'll merge it :-)

**Pull request merged**